src/openms/include/OpenMS/FORMAT/DATAACCESS/SiriusFragmentAnnotation.h

```diff
@@ -42,9 +42,35 @@ namespace OpenMS
       {
       public:
 
+          /**
+          @brief SiriusTargetDecoySpectra holds the target and/or decoy information for one entry (subdirectory from SIRIUS)
+          */
+          struct SiriusTargetDecoySpectra
+          {
+            MSSpectrum target;
+            MSSpectrum decoy;
+
+            SiriusTargetDecoySpectra() = default;
+            SiriusTargetDecoySpectra(MSSpectrum target_spectrum, MSSpectrum decoy_spectrum) : target(target_spectrum), decoy(decoy_spectrum) {}
+          };
+
+          /**
+          @brief extractAndResolveSiriusAnnotations
+          Extract and resolves SIRIUS target and/or decoy annotation for mapping native_id to MSSpectrum.
+
+          @return map native_id to annotated MSSpectrum (target or decoy)
+
+          If there are multiple identifications for a feature with the same MS2 spectras (concatenated native ids)
+          the identification with the higher SIRIUS score is chosen.
+
+          @param sirius_workspace_subdirs: Vector of paths to SIRIUS subdirectories.
+          @param use_exact_mass: Option to use exact mass instead of peak mz in MSSpectrum.
+          */
+          static std::vector<SiriusTargetDecoySpectra> extractAndResolveSiriusAnnotations(const std::vector<String>& sirius_workspace_subdirs, const double& score_threshold, bool use_exact_mass);
+
           /**
           @brief extractSiriusFragmentAnnotationMapping  
-          Extract native id (./spectrum.ms) and fragment annotation (./spectra/1_sumformula.ms) from SIRIUS output (per compound).
+          Extract concatenated native ids and concatenated mids (unique identifier) from (./spectrum.ms) and fragment annotation (./spectra/1_sumformula.tsv) from SIRIUS output (per compound).
```

Comment:
Is this "mids" or "m_ids"?

Reply:
See also other occurrences

Reply:
Should be mids, based on 
`std::vector<String> mids; ///< native ids and identifier for multiple possible identification via AMS ("|" separator)
`
In the SiriusMSConverter.h 

Where does the m_ids come from?

Reply:
I mean, it confuses me since I read it as "mids" = "middle points".
If it is a contraction of m... and IDs, maybe it is better called m_ids or mIDs

Reply:
ah, yeah that makes sense - will call it differently then!